### ISSUE TITLE
Installation Error

### ISSUE THREAD

**Issue by catubc** [DeepLabCut/DeepLabCut#300]
Hi
I was trying to install DLC and got this weird error in a brand-new DLC environment.  I first created a new env with python=3.6.8.

Perhaps distutils doesn't like python 3.6.8?

Any advice?

```
(base) cat@cathome:~/code/DeepLabCut$ activate dlc
(base) cat@cathome:~/code/DeepLabCut$ pip install deeplabcut
Collecting deeplabcut
  Downloading https://files.pythonhosted.org/packages/4d/f8/00492c68b51fb382340d40018733621323c6e5a4414b37e57e673eccb4b2/deeplabcut-2.0.6.3-py3-none-any.whl (140kB)
    100% |████████████████████████████████| 143kB 1.6MB/s 
Collecting matplotlib (from deeplabcut)
  Downloading https://files.pythonhosted.org/packages/dc/cb/a34046e75c9a4ecaf426ae0d0eada97078c8ce4bbe3250940b1a312a1385/matplotlib-3.1.0-cp37-cp37m-manylinux1_x86_64.whl (13.1MB)
    100% |████████████████████████████████| 13.1MB 1.8MB/s 
Collecting h5py~=2.7 (from deeplabcut)
  Downloading https://files.pythonhosted.org/packages/8e/fd/2ca5c4f4ed33ac4178f9c4d551e3946ab480866e3cd67a65a67a4bb35367/h5py-2.9.0-cp37-cp37m-manylinux1_x86_64.whl (2.8MB)
    100% |████████████████████████████████| 2.8MB 2.1MB/s 
Collecting intel-openmp (from deeplabcut)
  Downloading https://files.pythonhosted.org/packages/ea/f0/b1bc380d90241ab37149ed526a2ae585f6780ba616bd670b57f0559be37b/intel_openmp-2019.0-py2.py3-none-manylinux1_x86_64.whl (729kB)
    100% |████████████████████████████████| 737kB 2.5MB/s 
Collecting ipython~=6.0.0 (from deeplabcut)
  Downloading https://files.pythonhosted.org/packages/b9/87/1d7eba99c0108796ef643ac0e99d1d1dccf657bcff68e223bfb967c2512a/ipython-6.0.0-py3-none-any.whl (736kB)
    100% |████████████████████████████████| 737kB 1.6MB/s 
Collecting numpy~=1.14.5 (from deeplabcut)
  Downloading https://files.pythonhosted.org/packages/18/84/49b7f268741119328aeee0802aafb9bc2e164b36fc312daf83af95dae646/numpy-1.14.6-cp37-cp37m-manylinux1_x86_64.whl (13.8MB)
    100% |████████████████████████████████| 13.8MB 1.6MB/s 
Collecting scipy~=1.1.0 (from deeplabcut)
  Downloading https://files.pythonhosted.org/packages/40/de/0c22c6754370ba6b1fa8e53bd6e514d4a41a181125d405a501c215cbdbd6/scipy-1.1.0-cp37-cp37m-manylinux1_x86_64.whl (31.2MB)
    100% |████████████████████████████████| 31.2MB 985kB/s 
Requirement already satisfied: setuptools in /home/cat/miniconda3/lib/python3.7/site-packages (from deeplabcut) (41.0.0)
Requirement already satisfied: requests in /home/cat/miniconda3/lib/python3.7/site-packages (from deeplabcut) (2.21.0)
Collecting ruamel.yaml==0.15 (from deeplabcut)
  Downloading https://files.pythonhosted.org/packages/10/aa/f31c8a858e23fe5b14c59e6a03ae80b4cb76353c184df48a582a55eb2674/ruamel.yaml-0.15.0.tar.gz (247kB)
    100% |████████████████████████████████| 256kB 2.3MB/s 
    Complete output from command python setup.py egg_info:
    Warning: 'keywords' should be a list, got type 'NoneType'
    sys.argv ['-c', 'egg_info', '--egg-base', 'pip-egg-info']
    test compiling test_ruamel_yaml
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-install-yi4mg8x9/ruamel.yaml/setup.py", line 858, in <module>
        main()
      File "/tmp/pip-install-yi4mg8x9/ruamel.yaml/setup.py", line 847, in main
        setup(**kw)
      File "/home/cat/miniconda3/lib/python3.7/site-packages/setuptools/__init__.py", line 145, in setup
        return distutils.core.setup(**attrs)
      File "/home/cat/miniconda3/lib/python3.7/distutils/core.py", line 108, in setup
        _setup_distribution = dist = klass(attrs)
      File "/home/cat/miniconda3/lib/python3.7/site-packages/setuptools/dist.py", line 444, in __init__
        k: v for k, v in attrs.items()
      File "/home/cat/miniconda3/lib/python3.7/distutils/dist.py", line 267, in __init__
        getattr(self.metadata, "set_" + key)(val)
      File "/home/cat/miniconda3/lib/python3.7/distutils/dist.py", line 1203, in set_keywords
        self.keywords = _ensure_list(value, 'keywords')
      File "/home/cat/miniconda3/lib/python3.7/distutils/dist.py", line 40, in _ensure_list
        value = list(value)
    TypeError: 'NoneType' object is not iterable
    
    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-install-yi4mg8x9/ruamel.yaml/
```

**Comment by catubc** [comment]
UPDATE: Re-installed NVIDIA 384 driver and rebooted, now getting different error.  

https://www.dropbox.com/s/ow5k5h4ks0qwqm0/DLC_error_log.txt?dl=0

**Comment by AlexEMG** [comment]
In your first error, it seems that you didn't actually enter the environment (as the path is still referring to python3.7 ("File "/home/cat/miniconda3/lib/python3.7/distutils/dist.py").

As for the other error, we strongly recommend to use the provided environments:
https://github.com/AlexEMG/DeepLabCut/tree/master/conda-environments

It's not clear to me what your OS is, but you might need to install a few things like: ATLAS support etc. ```sudo apt-get install libatlas-base-dev```

**Comment by catubc** [comment]
Hi Alex. Thanks for the prompt response. I am in linux + conda, and here's what worked:

1.  I chose the 'import from cloned directory option' and that gave me a weird libculbas.so.10.0 error
`ImportError: libcublas.so.10.0: cannot open shared object file: No such file or directory
`
So that seemed to be related to an assumption that I have cuda toolkit 10 (which I don't on this workstation; and not sure it would work with older GPU).  So downgraded tensorflow-gpu to 1.12 to work with cudatoolkit 9.0 (NVIDIA 384.xx) and after manually installing all the missing packages, I was able to import deeplabcut directly through command-line python from the root directory of the cloned github dlc directory.  So thanks for the help with that.

2.  I also started the docker option, but I noticed that it uses some older tensorflow libraries and other older packages.  Perhaps if all else fails, I could return to that.

3.  Pip installing dlc is probalby still an option, my guess is that I need a more up-to-date cuda toolkit (and maybe a newer GPU!?).

Thanks for the help!
